### DISCUSSION TITLE
Free-Water Analysis Pipeline

### DISCUSSION THREAD

**Discussion by hookeba** [dipy/dipy#2633] [Reconstruction] [answered]
Hello DIPY experts!

Following the excellent free-water analysis tutorial and notes by @RafaelNetoHenr1, I believe the following should be sufficient for processing my brain-extracted data:

import numpy as np
import dipy.reconst.fwdti as fwdti
import dipy.reconst.dti as dti
import matplotlib.pyplot as plt
from dipy.io.image import load_nifti
from dipy.io.gradients import read_bvals_bvecs
from dipy.core.gradients import gradient_table
from dipy.io.image import save_nifti

data, affine = load_nifti('<DTI_DATA>.nii.gz')
bvals, bvecs = read_bvals_bvecs('bval', 'bvec')
gtab = gradient_table(bvals, bvecs)
fwdtimodel = fwdti.FreeWaterTensorModel(gtab)
fwdtifit = fwdtimodel.fit(data)

Before implementation have I missed any important steps?  In the tutorial code, I am uncertain about the following and how it may apply to my data:

data = np.asarray(img.dataobj)
affine = img.affine

Once again, thank you for your guidance.

Respectfully yours

@hookeba

**Comment by ShreyasFadnavis** [comment, marked as answer]
Hi @hookeba ! Sorry for the slowness on our end. The lines 
```
data = np.asarray(img.dataobj)
affine = img.affine
```

are equivalent to: `data, affine = load_nifti('<DTI_DATA>.nii.gz')`

In DIPY we have a wrapper around NiBabel which makes data handling easier :) Is there a specific error you are getting?
